README.md

```diff
@@ -1 +1,118 @@
 # xtractmime
+
+`xtractmime` is a [BSD-licensed](https://opensource.org/licenses/BSD-3-Clause)
+Python 3.6+ implementation of the [MIME Sniffing
+Standard](https://mimesniff.spec.whatwg.org/).
+
+Install from [`PyPI`](https://pypi.python.org/pypi/xtractmime):
+
+```
+pip install xtractmime
+```
+
+---
+
+## Basic usage
+
+Below mentioned are some simple examples of using `xtractmime.extract_mime`:
+
+```python
+>>> from xtractmime import extract_mime
+>>> body = b"Sample text content"
+>>> extract_mime(body)
+b'text/plain'
+>>>
+```
+
+```python
+>>> from xtractmime import extract_mime
+>>> content_types = (b'text/html',)
+>>> body = b''
+>>> extract_mime(body, content_types=content_types)
+b'text/html'
+>>> 
+```
+
+Additional functionality to check if a mime type belongs to a specific mime type group using 
+methods mentioned in `xtractmime.mimegroups`
+
+```python
+>>> from xtractmime.mimegroups import *
+>>> mime_type = b'text/html'
+>>> is_html_mime_type(mime_type)
+True
+>>> is_image_mime_type(mime_type)
+False
+>>>
+```
+
+---
+
+## API Reference
+
+### function `xtractmime.extract_mime : Optional[bytes]`
+
+Return `mimetype` as a byte string if the byte sequence in the provided content is recognized by `xtractmime`
+else return `None`
```

Comment:
`mimetype` is an implementation detail, the name of a variable used internally. Do not mention it in the documentation.

You also do not need to mention data types if they are already clear from the context (signature, parameters).

Make sure it is clear what the format of the output is, beyond the data type. For example, “Return the [MIME type essence](https://mimesniff.spec.whatwg.org/#mime-type-essence)” (link included) should make it clear that the returned bytes contain a MIME type in the form `<type>/<subtype>`. You could also follow with an example, to help people better understand the definition of the standard (or allow them not to read it in the first place): “(e.g. `text/html`)”.